### PR TITLE
upcoming: [M3-7874] - Linode Create Refactor - Marketplace - Part 1

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Begins building out the Marketplace section of the new Linode Create flow 🏗️ 

> [!note]
> I'm going to trial to do small consumable PRs for the remainder of the Linode Create refactor. If these PRs seem too small, just let me know.

## Preview 📷

Screenshot 2024-04-23 at 1 53 45 PM

### Prerequisites
- Turn on the `Linode Create v2` feature flag

### Verification steps
- Check http://localhost:3000/linodes/create?type=One-Click for parity with the existing Linode Create flow with what is build so far

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/Marketplace.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          />
+          <Autocomplete
+            textFieldProps={{
+              containerProps: { sx: { minWidth: 250 } },
+              hideLabel: true,
+            }}
+            label="Select category"
+            options={categoryOptions}
+            placeholder="Select category"
+          />
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/Marketplace.tsx:34]: Can we fix the slight height discrepancy that has always existed between these two inputs?

Screenshot 2024-04-23 at 7 56 23 PM

**Review by hkhalil-akamai — APPROVED** [review]
All for these short and sweet PRs! As long as it works for you

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/Marketplace.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/Marketplace.tsx:34]: I'll try taking a look, but if I can't figure it our cleanly, I'll make a ticket for this

File: packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/utilities.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  'Databases',
+  'Development',
+  'Games',
+  'Media and Entertainment',
+  'Monitoring',
+  'Productivity',
+  'Security',
+  'Stacks',
+  'Website',
+];
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/utilities.ts:12]: Can we generate this dynamically by doing 

```Typescript
const categories = oneClickApps.reduce((acc, app) => {
  return [...acc, ...app.categories];
}, [] as string[]);

const uniqueCategories = [...new Set(categories)]
```

or similar?

Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeCreatev2/Tabs/Marketplace/utilities.ts]: That is 1,000 IQ! 🧠 

Pushed in 4f592f81fc

**Review by abailly-akamai — APPROVED** [review]
Love the small PRs!

**Pull request merged**